File: split05/question05.py
class MyError(Exception):
    def __init__(self, data) -> None:
        self.data = data


    def __str__(self) -> str:
        return f'You input negative number: {self.data}. Try again.'
    

def check_positive(number) -> str: 
    try:
        if int(number) > 0:
            return f'You input positive number: {int(number)}'
        elif int(number) < 0:
            raise MyError(int(number))
    except MyError as message:
        return message
    except ValueError:
        return 'Error type: ValueError!'

File: split10/question05.js
function checkAdult(age) {
    try {
        if (age === undefined) {
            throw new Error("Please, enter your age");
        }
        else if (age < 0) {
            throw new Error("Please, enter positive number");
        }
        else if (!Number(age)) {
            throw new Error("Please, enter number");
        }
        else if (!Number.isInteger(age)) {
            throw new Error("Please, enter Integer number");
        }
        else if (age < 18) {
            throw new Error("Access denied - you are too young!");
        }
        else {
            console.log("Access allowed");
        }
    } catch(exception) {
        console.log(exception.name + " " + exception.message);
    } finally {
        console.log("Age verification complete");
    }
}